octobot_trading/exchanges/types/websocket_exchange.py

```diff
@@ -148,14 +153,34 @@ async def start_sockets(self):
     async def wait_sockets(self):
         await asyncio.wait(self.websocket_connectors_tasks)
 
-    async def close_and_restart_sockets(self):
-        for websocket in self.websocket_connectors:
-            await websocket.reset()
+    def _call_reset_ws_wrapper(self):
+        self.restarting = True
+        try:
+            for websocket in self.websocket_connectors:
+                asyncio_tools.run_coroutine_in_asyncio_loop(websocket.reset(), websocket.local_loop)
+        except Exception as e:
+            self.logger.exception(e, True, f"Error when resetting websockets {e}")
+        finally:
+            self.restarting = False
+
+    async def _inner_close_and_restart_sockets(self, debounce_duration):
```

Comment:
Are you sure we need this code here ? Why not in `CryptofeedWebSocketExchange` ?